### PR TITLE
Fix for Moodle 4.1

### PR DESCRIPTION
**Pull request by sarahjcotton**
Fixes issue #60

### CODE REVIEW

**Comment by brendanheywood** [comment]
ci is red on the master branch for this one

**Comment by sarahjcotton** [comment]
Currently fixing ci issues

**Comment by sarahjcotton** [comment]
All tests now passing - I've added missing docblocks the best I can but left out type hints so things don't break

File: classes/form/edit_conditions.php [hunk 1]
```diff
@@ -250,6 +268,8 @@ public function validation($data, $files) {
                     $errors["repeatvalue[$value]"] = get_string('formexperimentvalueerror', 'tool_abconfig');
                 }
                 // Increment total and check value.
+                // If an empty string is returned, set value to 0.
+                $data['repeatvalue'][$value] = $data['repeatvalue'][$value] ? '' : 0;
```
Comment (marxjohnson) [inline classes/form/edit_conditions.php]: This says "If `$data['repeatvalue'][$value]` is true, set it to an empty string, else set it to 0".

I think you mean something like
```php
if ($data['repeatvalue'][$value] == '') { 
    $data['repeatvalue'][$value] = 0;
}
```

Comment (sarahjcotton) [reply classes/form/edit_conditions.php]: This has been fixed so ready for re-review please @marxjohnson

**Comment by marxjohnson** [comment]
Thanks for that fix Sarah, all good now!

**Pull request merged**